ceph.md

```diff
@@ -381,6 +381,113 @@ spec:
+openstack user create --project service --password $SWIFT_PASSWORD swift
+openstack role create swiftoperator
+openstack role create ResellerAdmin
+openstack role add --user swift --project service member
+openstack role add --user swift --project service admin
+
+for i in public internal; do
+    openstack endpoint create --region regionOne object-store $i http://<RGW_ENDPOINT>:8080/swift/v1/AUTH_%\(tenant_id\)s;
+done
+
+openstack role add --project admin --user admin swiftoperator
+```
+
+- Replace `$SWIFT_PASSWORD` with the password that should be assigned to the
+  `swift` user.
+
+The following commands should be run on a Ceph server to deploy and configure
+a RGW service that will be able to serve object storage related requests.
+Before deploying the RGW instances, the first step is to make sure Ceph has the
+right `config-keys` required for the service to be able to interact with keystone.
+Add the following configuration to the Ceph cluster via `ceph config set` commands:
+
+```bash
+cephadm shell
+ceph config set global rgw_keystone_url "$KEYSTONE_ENDPOINT"
+ceph config set global rgw_keystone_verify_ssl false
+ceph config set global rgw_keystone_api_version 3
+ceph config set global rgw_keystone_accepted_roles "member, Member, admin"
+ceph config set global rgw_keystone_accepted_admin_roles "ResellerAdmin, swiftoperator"
+ceph config set global rgw_keystone_admin_domain default
+ceph config set global rgw_keystone_admin_project service
+ceph config set global rgw_keystone_admin_user swift
+ceph config set global rgw_keystone_admin_password "$SWIFT_PASSWORD"
+ceph config set global rgw_keystone_implicit_tenants true
+ceph config set global rgw_s3_auth_use_keystone true
+ceph config set global rgw_swift_versioning_enabled true
+ceph config set global rgw_swift_enforce_content_length true
+ceph config set global rgw_swift_account_in_url true
+ceph config set global rgw_trust_forwarded_https true
+ceph config set global rgw_max_attr_name_len 128
+ceph config set global rgw_max_attrs_num_in_req 90
+ceph config set global rgw_max_attr_size 1024
+```
+
+- Replace `$KEYSTONE_ENDPOINT` with the identity service public endpoint.
+- Replace `$SWIFT_PASSWORD` with the password assigned to the swift user in the
+  previous step.
+
+Deploy the RGW service and the associated ingress daemon:
+
+```bash
+cat <<EOF > "/tmp/rgw_spec"
+---
+service_type: rgw
+service_id: rgw
+service_name: rgw.rgw
+placement:
+  hosts:
+    - $HOST1
+    - $HOST2
+    ...
+    - $HOSTN
+networks:
+- $STORAGE_NETWORK
+spec:
+  rgw_frontend_port: 8082
+  rgw_realm: default
+  rgw_zone: default
+---
+service_type: ingress
+service_id: rgw.default
+service_name: ingress.rgw.default
+placement:
+  count: 1
+spec:
+  backend_service: rgw.rgw
+  frontend_port: 8080
+  monitor_port: 8999
+  virtual_ip: $STORAGE_NETWORK_VIP
+  virtual_interface_networks:
+  - $STORAGE_NETWORK
+```
+
+- Replace `$HOST1`, `$HOST2`, ..., `$HOSTN` with the number of `CephStorage`
```

Comment:
Let's not use `CephStorage` because that had special meaning in TripleO and we don't want anyone to think some form of it has lived on in the new deployment method.

"with the number of Ceph nodes where the ..."

Reply:
Ack I'm going to `s/CephStorage/Ceph` here. Let's avoid confusion or any call to other projects.

Reply:
Done